lib/vauth/ntlm.c

```diff
@@ -845,8 +825,13 @@ CURLcode Curl_auth_create_ntlm_type3_message(struct Curl_easy *data,
   if(result)
     return CURLE_CONV_FAILED;
 
-  /* Return with binary blob encoded into base64 */
-  result = Curl_base64_encode(data, (char *)ntlmbuf, size, outptr, outlen);
+  /* Return the binary blob. */
+  cp = malloc(size + 1);
+  if(!cp)
+    return CURLE_OUT_OF_MEMORY;
+  memcpy(cp, ntlmbuf, size);
+  cp[size] = '\0';
```

Comment:
why do you zero-terminate this buffer? It's a binary blob, right? `Curl_memdup()` can be used otherwise.

Reply:
I have always included a zero-terminator as a sentinel (as `Curl_base64_decode` does) in case data is wronly used later with string functions.